pinc/LPage.inc

```diff
@@ -310,6 +310,21 @@ class LPage
         return $page_text;
     }
 
+    public function get_info()
+    {
+        foreach ($this->round->other_rounds_with_visible_usernames as $other_round_id) {
+            $other_round = get_Round_for_round_id($other_round_id);
+            $username = $this->get_username_for_round($other_round);
+            $user_id = get_forum_user_id($username);
+            $round_info_array[] = [$other_round->id, $username, $user_id];
+        }
+
+        return [
```

Comment:
I'd be inclined to return either an associative array with labeled keys or an adhoc stdClass object with named attributes. Right how this is a very opaque structure that requires a lot of mental context to trace:
```
[
    <imagename>,
    [
        [<roundid>, <username>, <userid>],
        ...
    ]
]
```

compared to:
```
[
    "imagename": <imagename>,
    "round_info": [
        [
            "round_id": <roundid>,
            "username": <username>,
            "user_id <user_id>,
        ],
        ...
    ]
]
```

I'm curious on @bpfoley's thoughts on this too.

Reply:
It would probably be clearer if we broke this out into two functions: one that returned the pagename and one that returned the round details.

Reply:
Yeah. If this was Python, I'd say "use a dataclass instead of a list". In PHP 8 we can approximate this with (from memory)
```php
class RoundInfo
{
  public function __construct(
    public string $round_id,
    public string $username,
    public int $user_id
  ) {}
}
```
but since it's PHP 7.4 we can't do this neatly yet, so I'd say using `stdClass` (there's an example in `get_OCR_pseudoround`) is as close as we can get for now.

Reply:
Classes are probably more efficient than associative arays because the keys are not duplicated for every instance [see https://www.exakat.io/en/moving-from-array-to-class/](https://www.exakat.io/en/moving-from-array-to-class/). But as Casey says, clarity of the code is the most important thing. I'll try both ways and see how they look.